cli/cliui/agent.go

```diff
@@ -257,6 +257,25 @@ func troubleshootingMessage(agent codersdk.WorkspaceAgent, url string) string {
 	return m
 }
 
+// safeDuration returns a-b. If a or b is nil, it returns 0.
+// This is because we often dereference a time pointer, which can
+// cause a panic. These dereferences are used to calculate durations,
+// which are not critical, and therefor should not break things
+// when it fails.
+// A panic has been observed in a test.
+func safeDuration(sw *stageWriter, a, b *time.Time) time.Duration {
+	if a == nil || b == nil {
+		if sw != nil {
```

Comment:
No need for sw to be an optional param here, IMO 🤔 (i.e. no need to check).

Reply:
I just really didn't want to accidently introduce a new place to deref a nil lol